### ISSUE TITLE
Implement transaction prevalidation

### ISSUE THREAD

**Issue by tuand27613** [hyperledger-archives/fabric#482]
as noted in #273 
/*see e.g. obc-classic.go  verify() which is currently stubbed out , waiting for TransactionPreValidation() */
edited:  TransactionPrevalidate() actually done at consensus.RecvMsg() level.  This verify is for syntax/semantics validation of pbft messages at innerCPI level.
Also, rename to validate() per @corecode

**Comment by kchristidis** [comment]
@tuand27613 Can you please post a status update for this?

Also, it'd be great if you could update the original item description by providing a detailed (enough) description of what needs to be done. Link to the code and use code snippets (please use backticks for `code`) where applicable. Provide the necessary context. The original description is hard to parse. This should read like a min-report of sorts that gives others a complete idea of what's going on. See @jyellick's posts and updates on #291 for an example.

**Comment by tuand27613** [comment]
from @corecode : 
- verify() should be renamed to validate() as it is really meant for doing syntax/semantics checking on pbft messages. This is done as part of #273  
- This validation is meant mainly for sieve and are no-ops for classic and batch. The validation code is already implemented in obc-sieve. See, for example, `func (op *obcSieve) validate(rawReq []byte) error`  which in turn calls `func (op *obcSieve) validateVerifySet(vset *VerifySet) error` and `func (op *obcSieve) validateFlush(flush *Flush) error`. In obc-classic and obc-batch, the functions are stubbed (return nil). 
  So for now, there is nothing more to do.  There are possible edge cases that might require extra validation in batch and classic but we have not started discussing these yet. We'll keep this issue open to record these discussions.
